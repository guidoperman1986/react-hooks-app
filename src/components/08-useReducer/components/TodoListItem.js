import React from 'react'

export const TodoListItem = ({todo, i, handleDelete, handleToggle}) => {
    return (
        <li className="list-group-item text-center" 
            key={todo.id}
            onClick={ () => handleToggle(todo.id) }
            > 
            <p className={ `${todo.done && 'complete'}` }>
                {i + 1}. { todo.desc }
            </p>
            <button 
                className="btn btn-danger" 
                onClick={ ()=>handleDelete(todo.id) }
            >Borrar</button>
        </li>
    )
}
