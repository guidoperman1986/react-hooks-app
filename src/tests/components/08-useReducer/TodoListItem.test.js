import React from 'react';
import { shallow } from "enzyme";
import { TodoListItem } from '../../../components/08-useReducer/components/TodoListItem';

import { demoTodos } from '../../fixtures/demoTodos'

describe('Pruebas en TodoListItem', () => {

    const handleDelete = jest.fn();
    const handleToggle = jest.fn();

    const wrapper = shallow(<TodoListItem todo={demoTodos[0]} i={0} handleDelete={handleDelete} handleToggle={handleToggle} />)

    test('should create', () => {
        expect(wrapper).toMatchSnapshot();
        
    })

    test('should call handleDelete', () => {
        wrapper.find('button').simulate('click');
        expect(handleDelete).toHaveBeenCalledWith(1);
    })

    test('should call handleToggle', () => {
        wrapper.find('li').simulate('click');
        expect(handleToggle).toHaveBeenCalledWith(1);
    })

    test('should show text correctly', () => {
        expect(wrapper.find('p').text()).toBe(`1. ${ demoTodos[0].desc }`)
    })

    test('should show linethrough', () => {
        const todo = demoTodos[0];
        todo.done = true;

        const wrapper = shallow(<TodoListItem todo={todo} />)

        expect(wrapper.find('p').hasClass('complete')).toBeTruthy();

    })


    
    
    
    
})


