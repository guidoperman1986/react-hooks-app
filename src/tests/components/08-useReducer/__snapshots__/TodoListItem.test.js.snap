// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en TodoListItem should create 1`] = `
<li
  className="list-group-item text-center"
  key="1"
  onClick={[Function]}
>
  <p
    className="false"
  >
    1
    . 
    Aprender React
  </p>
  <button
    className="btn btn-danger"
    onClick={[Function]}
  >
    Borrar
  </button>
</li>
`;
